File: C/quicksort_secuencial.c
// Programa para buscar un elemento de forma secuencial y ordenarlo mediante quicksort

#include <stdio.h>

// Funcion para llenar un arreglo

void fill_array(int arr[], int size)
{
    register int i;
    int element;

    for (i = 0; i < size; i++)
    {
        printf("\nIngrese el valor del elemento %d ", i);
        scanf("%d", &element);
        arr[i] = element;
    }
    printf("\n");
}

// Funcion para imprimir un arreglo

void print_array(int arr[], int size)
{
    register int i;

    printf("\nLos elementos del arreglo son: [ ");
    
    for (i = 0; i < size; i++)
    {
        printf("%d ",arr[i]);
    }
    
    printf("]\n");
}

// Funcion para implementar la busqueda secuencial

int linear_search(int arr[], int size, int v)
{
    register int i;

    for (i = 0; i < size; i++)
    {
        if (arr[i] == v)
            return i;
    }

    return -1;
}

// Funcion para particionar el arreglo

int partition(int arr[], int start, int end)
{
    register int j;

    int pivote = arr[end];
    int aux;
    int i = start - 1;

    for (j = start; j <= end - 1; j++)
    {
        if (arr[j] < pivote)
        {
            i++;
            aux = arr[i];
            arr[i] = arr[j];
            arr[j] = aux;
        }
    }
    i++;
    aux = arr[i];
    arr[i] = arr[end];
    arr[end] = aux;

    return i;
}

// Funcion para el ordenamiento por quicksort

void quick_sort(int arr[], int start, int end)
{
    int pivote;

    if (start < end)
    {
        pivote = partition(arr, start, end);
        quick_sort(arr, start, pivote - 1);
        quick_sort(arr, pivote + 1, end);
    }
}

void main()
{
    int n, value, pos;

    printf("\nIngrese la cantidad de elementos del arreglo: ");
    scanf("%d", &n);

    int array[n];

    // Llamada a funcion para rellenar el arreglo

    fill_array(array, n);
    print_array(array, n);

    // Llamada a funcion para buscar un elemento de forma secuencial

    printf("\nIngrese el valor del elemento a buscar: ");
    scanf("%d", &value);

    pos = linear_search(array, n, value);
    printf("\nLa posicion del elemento %d es: %d", value, pos);

    // Llamada a funcion para ordenar por quicksort

    printf("\nOrdenando mediante quicksort...\n");
    quick_sort(array, 0, n - 1);
    print_array(array, n);
}

File: Python/quicksort_secuencial.py
# Programa para buscar un elemento de forma secuencial y ordenar el arreglo mediante quicksort

# Funcion para rellenar una lista

def fill_list(lista):
    length = int(input("\nIngrese la longitud de la lista: "))

    for i in range(0, length):
        element = int(input(f"\nIngresa el valor del elemento {i}: "))
        lista.append(element)

# Funcion para la linear search

def linear_search(lista, size, v):

    for i in range(0, size):
        if lista[i] == v:
            return i

    return -1

# Funcion para particionar la lista

def partition(lista, start, end):

    pivote = lista[end]
    i = start

    for j in range(start, end):
        if lista[j] < pivote:
            aux = lista[i]
            lista[i] = lista[j]
            lista[j] = aux
            i += 1
    
    aux = lista[i]
    lista[i] = lista[end]
    lista[end] = aux

    return i

# Funcion para el ordenamiento por quicksort

def quick_sort(lista, start, end):

    if start < end:
        pivote = partition(lista, start, end)
        quick_sort(lista, start, pivote - 1)
        quick_sort(lista, pivote + 1, end)



# Programa main()

int_list = []

# Llamada a funcion para rellenar una lista

fill_list(int_list)
print(f"\nLa lista es: {int_list}")

# Calculo de la longitud de la lista

list_size = len(int_list)

# Llamada a funcion para buscar un elemento de forma secuencial

value = int(input("Ingrese el valor del elemento a buscar: "))
pos = linear_search(int_list, list_size, value)
print(f"\nLa posicion del elemento {value} es: {pos}")

# Llamada a funcion para ordenar mediante quicksort

print("\nOrdenando la lista por quicksort...\n")
quick_sort(int_list, 0, list_size - 1)
print(f"\nLa lista es: {int_list}")